pkg/asset/installconfig/basedomain.go

```diff
@@ -48,6 +50,12 @@ func (a *baseDomain) Generate(parents asset.Parents) error {
 		}
 		a.BaseDomain = zone.Name
 		return platform.Azure.SetBaseDomain(zone.ID)
+	case gcp.Name:
+		var err error
```

Comment:
https://github.com/openshift/installer/blob/71a463832b0cc7a60b52b2224c07e32273f54afb/pkg/asset/installconfig/basedomain.go#L36-L41

The reasoning for AWS adopting if rate limited or forbidden or none-found, fallback to string input for base domain is that during `create install-config` the install-config doesn't require the DNS zone's cloud id, that's a requirement for different assets and therefore, a string base domain is also acceptable.

I think it's acceptable to follow that behavior.